### PR TITLE
Problem: no instructions for compiling chain-core to WASM#1690

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: add markdown

### CODE REVIEW

File: chain-core/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// this is dummy function to build WASM
+fn main()
+{
+    println!("chain-core wasm ok");
+}
```
Comment (tomtau) [inline chain-core/src/main.rs:5]: not need to put this in chain-core -- the readme doc can just have an example crate with main.rs that imports chain-core and does some simple stuff (creates a signed transaction? prints tx id?) and one can compile and run in WASM

File: chain-core/BUILD-WASM.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+----------------------------
+
+## install emcc for compile WASM
+1. use ubuntu 18.x
+2. git clone https://github.com/emscripten-core/emsdk.git
+3. cd emsdk
+4. ./emsdk install latest
+5. ./emsdk activate latest
+6. source ./emsdk_env.sh
+
```
Comment (tomtau) [inline chain-core/BUILD-WASM.md:11]: for macOS? `brew install emscripten`?

File: chain-core/BUILD-WASM.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+add to dependencies
+```
+secp256k1 = ""
+rand=""
+```
+
+## build
+1. cargo build --target=wasm32-unknown-emscripten
+2. cd target/wasm32-unknown-emscripten/debug/
```
Comment (tomtau) [inline chain-core/BUILD-WASM.md:41]: no need for this?

File: chain-core/src/main.rs (hunk 1 above)
Comment (leejw51crypto) [reply chain-core/src/main.rs:5]: agree, i'll fix it

File: chain-core/BUILD-WASM.md (hunk 2 above)
Comment (leejw51crypto) [reply chain-core/BUILD-WASM.md:11]: for mac, i'll try

File: chain-core/BUILD-WASM.md (hunk 3 above)
Comment (leejw51crypto) [reply chain-core/BUILD-WASM.md:41]: ok, i'll remove it

**Comment by leejw51crypto** [comment]
for mac, not working
i need to verify it in hello-world level.
how about as another issue?

**Comment by leejw51crypto** [comment]
trying to compile on mac , now

**Comment by leejw51crypto** [comment]
update guide for mac,
with fix of ring, WASM compiles fine

File: chain-core/BUILD-WASM.md (hunk 2 above)
Comment (leejw51crypto) [reply chain-core/BUILD-WASM.md:11]: mac working fine, added to markdown for additional setup

**Comment by leejw51crypto** [comment]
need to modify ring, i'll submit as another PR to ring

File: chain-core/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    tx.add_output(TxOut::new(ExtendedAddr::OrTree([0xbb; 32]), Coin::unit()));
+    let secp = Secp256k1::new();
+    let sk1 = SecretKey::from_slice(&[0xcc; 32][..]).expect("secret key");
+    let pk1 = PublicKey::from_secret_key(&secp, &sk1);
+    let raw_pk1 = RawXOnlyPubkey::from(XOnlyPublicKey::from_pubkey(&pk1).0.serialize());
+    println!("pubkey: {:?}", raw_pk1);
+    let msg = Message::from_slice(&tx.id()).expect("msg");
+    let sig = schnorr_sign(&secp, &msg, &sk1);
+    println!("signature: {:?}", sig);
+}
```
Comment (tomtau) [inline chain-core/src/main.rs:24]: no `main.rs` for chain-core

Comment (leejw51crypto) [reply chain-core/src/main.rs:24]: ok, i'll change it

**Pull request merged**